app/src/main/java/com/android/streetworkapp/ui/utils/CustomDialog.kt

```diff
@@ -84,10 +63,106 @@ fun CustomDialog(
 }
 
 enum class DialogType {
-  QUERY, // Have submit / cancel buttons + use OnSubmit / onDismiss.
-  INFO; // Only display the content + use onDismiss.
+  QUERY, // Have {submit, cancel} buttons + use {onSubmit, onDismiss}.
+  INFO, // Only display the content + use onDismiss.
+  CONFIRM; // Display the content + {confirm} button + {onSubmit, onDismiss}.
+
+  fun shouldShowButton(button: ButtonType): Boolean {
+    return when (button) {
+      ButtonType.SUBMIT -> this == QUERY
+      ButtonType.CANCEL -> this == QUERY
+      ButtonType.CONFIRM -> this == CONFIRM
+    }
+  }
+}
+
+/** Enum class for the button types. Each type has a response, tag, and color. */
+enum class ButtonType {
+  SUBMIT,
+  CANCEL,
+  CONFIRM;
+
+  fun response(context: Context): String {
+    return when (this) {
+      SUBMIT -> context.getString(R.string.DialogSubmitButtonResponse)
+      CANCEL -> context.getString(R.string.DialogCancelButtonResponse)
+      CONFIRM -> context.getString(R.string.DialogConfirmButtonResponse)
+    }
+  }
+
+  fun tag(context: Context): String {
+    return when (this) {
+      SUBMIT -> context.getString(R.string.DialogSubmitButtonTitle)
+      CANCEL -> context.getString(R.string.DialogCancelButtonTitle)
+      CONFIRM -> context.getString(R.string.DialogConfirmButtonTitle)
+    }
+  }
+
+  fun color(): Color {
+    return when (this) {
+      CONFIRM -> Color.Red
+      CANCEL -> Color.Red
+      else -> ColorPalette.SECONDARY_TEXT_COLOR
+    }
+  }
+}
+
+/**
+ * This function displays the button for the dialog. Will hide the dialog once interacted with.
+ *
+ * @param ButtonType - Type of button (SUBMIT, CANCEL, CONFIRM)
+ * @param tag - Describe the instance of the dialog ("Settings", "Rating", etc.)
+ * @param onSubmit - Submission callback
+ * @param showDialog - MutableState to show the dialog
+ * @param verbose - Whether to display a Toast message on interaction
+ */
+@Composable
+fun DialogButton(
+    ButtonType: ButtonType,
```

Comment:
Sonar raised an issue about the parameter naming, please do not use a capital letter for the first letter.

Reply:
The changes have been pushed